test/e2e/elasticsearch_token_propagation_test.go

```diff
@@ -0,0 +1,337 @@
+	/* Create an span */
+	collectorPort := randomPortNumber()
+	collectorPorts := []string{collectorPort + ":14268"}
+	portForwColl, closeChanColl := CreatePortForward(namespace, suite.collectorName, collectorPodImageName, collectorPorts, fw.KubeConfig)
+	defer portForwColl.Close()
+	defer close(closeChanColl)
+	collectorEndpoint := fmt.Sprintf("http://localhost:%s/api/traces", collectorPort)
+	cfg := config.Configuration{
+		Reporter:    &config.ReporterConfig{CollectorEndpoint: collectorEndpoint},
+		Sampler:     &config.SamplerConfig{Type: "const", Param: 1},
+		ServiceName: testServiceName,
+	}
+	tracer, closer, err := cfg.NewTracer()
+	require.NoError(t, err, "Failed to create tracer in token propagation test")
+	tStr := time.Now().Format(time.RFC3339Nano)
+	tracer.StartSpan("TokenTest").
+		SetTag("time-RFC3339Nano", tStr).
+		Finish()
+	closer.Close()
+	client := newHTTPSClient()
+	/* Try to reach query endpoint */
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		req, err := http.NewRequest(http.MethodGet, suite.queryServiceEndPoint, nil)
+		req.Header.Add("Authorization", "Bearer "+suite.token)
+		resp, err := client.Do(req)
+		defer resp.Body.Close()
+		require.Equal(t, http.StatusOK, resp.StatusCode)
+		if resp.StatusCode != http.StatusOK {
+			return false, errors.New("Query service returns http code: " + string(resp.StatusCode))
+		}
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		bodyString := string(bodyBytes)
+		if !strings.Contains(bodyString, "errors\":null") {
+			return false, errors.New("query service returns errors: " + bodyString)
+		}
+		return true, nil
+	})
+	require.NoError(t, err, "Token propagation test failed")
+}
+
+func (suite *TokenTestSuite) deployJaegerWithPropagationEnabled() {
+	queryName := fmt.Sprintf("%s-query", name)
+	collectorName := fmt.Sprintf("%s-collector", name)
+	bindOperatorWithAuthDelegator()
+	createTestServiceAccount()
+	suite.token = testAccountToken()
+
+	suite.exampleJaeger = jaegerInstance()
+	err := fw.Client.Create(goctx.Background(),
+		suite.exampleJaeger,
+		&framework.CleanupOptions{
+			TestContext:   ctx,
+			Timeout:       timeout,
+			RetryInterval: retryInterval,
+		})
+	require.NoError(t, err, "Error deploying example Jaeger")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, collectorName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for collector deployment")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, queryName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for query deployment")
+
+	route := findRoute(t, fw, name)
+
+	suite.host = route.Spec.Host
+	suite.queryServiceEndPoint = fmt.Sprintf("https://%s/api/traces?service=%s", suite.host, testServiceName)
+}
+
+func TestTokenSuite(t *testing.T) {
+	suite.Run(t, new(TokenTestSuite))
+}
+
+func bindOperatorWithAuthDelegator() {
+	roleBinding := rbac.ClusterRoleBinding{
```

Comment:
This clusterrolebinding and the token-test-user-bind clusterrolebinding created below need to be deleted at the end of the test (in TearDownSuite()) otherwise the test will fail after the first run as these will already exist on the cluster.

Reply:
Thanks for the observation, I've added cleaning routine for remove all created roles/role-bindings.